Detectors/EMCAL/workflow/src/StandaloneAODProducerSpec.cxx

```diff
@@ -0,0 +1,173 @@
+#include "Framework/ConfigParamRegistry.h"
+#include "Framework/DataTypes.h"
+#include "Framework/InputRecordWalker.h"
+#include "Framework/Logger.h"
+#include "Framework/TableBuilder.h"
+#include "Framework/TableTreeHelpers.h"
+#include "MathUtils/Utils.h"
+
+using namespace o2::framework;
+
+namespace o2
+{
+namespace emcal
+{
+
+StandaloneAODProducerSpec::StandaloneAODProducerSpec()
+{
+  mTimer.Stop();
+  mTimer.Reset();
+}
+
+void StandaloneAODProducerSpec::init(o2::framework::InitContext& ic)
+{
+  mCaloEventHandler = new o2::emcal::EventHandler<o2::emcal::Cell>();
+  mCaloAmp = 0xFFFFFFFF;
+  mCaloTime = 0xFFFFFFFF;
+
+  mTFNumber = ic.options().get<int64_t>("aod-timeframe-id");
+  mRunNumber = ic.options().get<int>("run-number");
+}
+
+void StandaloneAODProducerSpec::run(ProcessingContext& pc)
+{
+  auto cput = mTimer.CpuTime();
+  mTimer.Start(false);
+
+  const auto* dh = o2::header::get<o2::header::DataHeader*>(pc.inputs().getFirstValid(true).header);
+  uint64_t tfNumber;
+  if (mTFNumber == -1L) {
+    // TODO has to be made globally unique (by using absolute time of TF). For now is unique within the run
+    tfNumber = dh->tfCounter; // getTFNumber(startIR, runNumber);
+  } else {
+    tfNumber = mTFNumber;
+  }
+  const int runNumber = (mRunNumber == -1) ? int(dh->runNumber) : mRunNumber;
+
+  auto cellsIn = pc.inputs().get<gsl::span<o2::emcal::Cell>>(getCellBinding());
+  auto triggersIn = pc.inputs().get<gsl::span<o2::emcal::TriggerRecord>>(getCellTriggerRecordBinding());
+
+  LOG(INFO) << "FOUND " << cellsIn.size() << " EMC cells in CTF";
+  LOG(INFO) << "FOUND " << triggersIn.size() << " EMC tiggers in CTF";
+
+  auto& bcBuilder = pc.outputs().make<TableBuilder>(Output{"AOD", "BC"});
+  auto& collisionsBuilder = pc.outputs().make<TableBuilder>(Output{"AOD", "COLLISION"});
+  auto& caloCellsBuilder = pc.outputs().make<TableBuilder>(Output{"AOD", "CALO"});
+  auto& caloCellsTRGTableBuilder = pc.outputs().make<TableBuilder>(Output{"AOD", "CALOTRIGGER"});
+
+  auto bcCursor = bcBuilder.cursor<o2::aod::BCs>();
+  auto collisionsCursor = collisionsBuilder.cursor<o2::aod::Collisions>();
+  auto caloCellsCursor = caloCellsBuilder.cursor<o2::aod::Calos>();
+  auto caloCellsTRGTableCursor = caloCellsTRGTableBuilder.cursor<o2::aod::CaloTriggers>();
+
+  // build event for easier handling
+  mCaloEventHandler->reset();
+  mCaloEventHandler->setCellData(cellsIn, triggersIn);
+
+  std::vector<int64_t> bcMap;
+  // loop over events
+  for (int iev = 0; iev < mCaloEventHandler->getNumberOfEvents(); iev++) {
+    o2::emcal::EventData inputEvent = mCaloEventHandler->buildEvent(iev);
+    auto cellsInEvent = inputEvent.mCells;                  // get cells belonging to current event
+    auto interactionRecord = inputEvent.mInteractionRecord; // get interaction records belonging to current event
+
+    LOG(INFO) << "Found " << cellsInEvent.size() << " cells in event";
+
+    auto bcID = interactionRecord.toLong();
```

Comment:
This is wrong. You have to fill into the tables the tree entry of the BC entry in the BC tree. You get this with `bcCursor.lastIndex()`

Reply:
`bcCursor` is a lambda function, `lastIndex()` doesn't exist. Can't we just take a running index (i.e. iev) that we increment per event/bc in the loop?